### PR TITLE
Binary classifier

### PR DESCRIPTION
**Pull request by zehdari**
## Summary by CodeRabbit

* **New Features**
  * Added binary-classifier-driven mapping with two-instance clustering/locking and mapping updates based on the selected instance.
  * Added services to start, freeze, start the second instance, and stop the classifier.
  * Expanded 3D vision post-processing (including slalom, gate pairing, and torpedo separation) with generated markers and point clouds.
* **Configuration Updates**
  * Added a dedicated binary-classifier parameter set and expanded mapping targets (including `bin`, `bin_target1`, `bin_target2`).
  * Updated initial frame/object definitions and improved dummy-detection coverage (including new blood objects and orientation-invalid handling).
  * Updated YOLO orientation model/label mappings and added `use_sim_time` support.

### CODE REVIEW

**Comment by zehdari** [comment]
@coderabbitai full review

File: riptide_mapping/riptide_mapping2/binary_classifier.py [hunk 1]
```diff
@@ -0,0 +1,446 @@
...
+            return
+
+        # drop samples older than the TTL so old views of a bin don't linger after we've moved on
+        self.samples = deque(
+            [sample for sample in self.samples if now_sec - sample.stamp_sec <= self.params.buffer_ttl_sec],
+            maxlen=max(1, self.params.buffer_size),
+        )
+
+    def buffer_size(self):
+        return self.samples.c
```
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/binary_classifier.py]: This is dead code, unused and needs deleted

File: riptide_mapping/riptide_mapping2/binary_classifier.py [hunk 2]
```diff
@@ -0,0 +1,446 @@
...
+
+    def param_name(self, name):
+        return "{}.{}".format(self.param_namespace, name)
+
+    def declare_params(self):
+        # defaults come from the shared DEFAULTS dict; the launch-provided yaml overrides them
+        self.node.declare_parameters(
+            namespace="",
+            parameters=[(self.param_name(name), value) for name, value in DEFAULTS.items()],
+        )
```
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/binary_classifier.py:111]: Won't really happen in practice, but good for non-node instantiated unit tests

File: riptide_mapping/riptide_mapping2/mapping.py [hunk 3]
```diff
@@ -183,14 +247,145 @@ def target_callback(self, request: MappingTarget.Request, response: MappingTarge
...
+    def start_binary_classifier_second_callback(self, request: Trigger.Request, response: Trigger.Response):
+        success, message = self.binary_classifier.start_second()
+
+        if success:
+            # If start_second locked instance 2 from the (preserved) buffer, publish it now as a TF for autonomy to navigate to, 
+            # but high covariance so it isn't treated as confirmed.
+            if self.binary_classifier.second_locked:
+                self.seed_object_estimate(self.binary_classifier.instance2_name, self.binary_classifier.instance2_centroid)
+                self.publish_pose()
+            else:
```
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/mapping.py:373]: This one could be a real issue, need to look into this/test on the robot

[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/mapping.py:373]: @coderabbitai open a follow-up issue for this

**Review by zehdari — COMMENTED** [review]
Tested Tirpak test, big W

**Comment by zehdari** [comment]
@coderabbitai full review

File: riptide_mapping/config/dummy_detections.yaml [hunk 4]
```diff
@@ -1,6 +1,26 @@
...
+      - gate
+      - gate_shark
+      - gate_saw
+      - slalom_front
+      - slalom_middle
+      - slalom_back
+      - torpedo
+      - torpedo_shark_hole
+      - torpedo_sawfish_hole
+      - bin_target
```
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/config/dummy_detections.yaml:24]: @AdityaGupta0 you can fix this one for tree testing

File: riptide_mapping/riptide_mapping2/mapping.py [hunk 5]
```diff
@@ -264,23 +539,26 @@ def try_update_pose(self, result: ObjectHypothesisWithPose, detection_header: He
...
-        update_position = True
-        update_orientation = True
+            return False, None, None, str(ex)
 
         trans_pose = do_transform_pose_stamped(pose, transform)
-        
-        
-        if result.hypothesis.class_id in self.downwards_objects.keys(): # and parent == "map":
+        if result.hypothesis.class_id in self.downwards_objects.keys():
             trans_pose.pose.orientation.x = 0.0
```
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/mapping.py]: Probably will do, needs tested in sim/bench

File: riptide_mapping/riptide_mapping2/dummydetections.py [hunk 6]
```diff
@@ -63,7 +52,7 @@ def __init__(self):
         self.pubs        = [ ]
         self.srv = self.create_service(SetBool, 'set_camera_is_dfc', self.setActiveCameraCb)
 
-        for object in objects:
+        for object in self.objects:
             self.pubs.append(self.create_publisher(PoseWithCovarianceStamped, f"dummydetections/{object}", 10))
```
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/dummydetections.py]: @AdityaGupta0 you can do this when making dummy dets changes as well, not like we use a linter yet anyways lol

File: tensor_detector/src/geometry.py [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+    axis = np.cross(default_normal, normal)
+    axis_length = np.linalg.norm(axis)
+    if axis_length == 0:
+        # Normal is parallel/anti-parallel, rotate 180 deg about an arbitrary axis (goofy ahh rotation)
+        axis = np.array([1, 0, 0])
+        angle = np.pi
+    else:
+        axis = axis / axis_length
+        angle = np.arccos(np.dot(default_normal, normal))
+    return R.from_rotvec(axis * angle)
```
[the remark this answers is not in the document]
Comment (zehdari) [reply tensor_detector/src/geometry.py]: Maybe, I'll test but seems like an edge case. Technically normal_to_quaternion already guards this but may be good if we eventually (ever) do unit tests

File: riptide_mapping/riptide_mapping2/mapping.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (zehdari) [reply riptide_mapping/riptide_mapping2/mapping.py]: @coderabbitai make an issue for this

File: tensor_detector/src/geometry.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (zehdari) [reply tensor_detector/src/geometry.py]: @coderabbitai make an issue for this

File: riptide_mapping/riptide_mapping2/dummydetections.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (AdityaGupta0) [reply riptide_mapping/riptide_mapping2/dummydetections.py]: @coderabbitai check it

File: riptide_mapping/config/dummy_detections.yaml (hunk 4 above)
[the remark this answers is not in the document]
Comment (AdityaGupta0) [reply riptide_mapping/config/dummy_detections.yaml:24]: @coderabbitai check it

[the remark this answers is not in the document]
Comment (AdityaGupta0) [reply riptide_mapping/config/dummy_detections.yaml:24]: fixed the nit
